### PR TITLE
Add Roc Curve and Precision/Recall Curve to the metrics

### PR DESCRIPTION
**Pull request by GabrielePicco**
Description:
Add Roc Curve and Precision/Recall Curve to the metrics

Check list:
* [x] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format

### CODE REVIEW

File: ignite/contrib/metrics/precision_recall_curve.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    `sklearn.metrics.precision_recall_curve <http://scikit-learn.org/stable/modules/generated/
+    sklearn.metrics.precision_recall_curve.html#sklearn.metrics.precision_recall_curve>`_ .
+
+    Args:
+        output_transform (callable, optional): a callable that is used to transform the
+            :class:`~ignite.engine.Engine`'s `process_function`'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+
+    ROC_AUC expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```
Comment (vfdev-5) [inline ignite/contrib/metrics/precision_recall_curve.py:27]:
```suggestion
    PrecisionRecallCurve expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```

File: ignite/contrib/metrics/roc_curve.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    `sklearn.metrics.roc_curve <http://scikit-learn.org/stable/modules/generated/
+    sklearn.metrics.roc_curve.html#sklearn.metrics.roc_curve>`_ .
+
+    Args:
+        output_transform (callable, optional): a callable that is used to transform the
+            :class:`~ignite.engine.Engine`'s `process_function`'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+
+    ROC_AUC expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```
Comment (vfdev-5) [inline ignite/contrib/metrics/roc_curve.py:27]:
```suggestion
    RocCurve expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```

**Comment by GabrielePicco** [comment]
Thanks @vfdev-5 for the review! I added the metrics to the module and fixed the typos.
Having them in two separate files seems clearer to me, but if you think it's better I can join them.

**Comment by vfdev-5** [comment]
> but if you think it's better I can join them.
@GabrielePicco yes, please ;)

**Comment by GabrielePicco** [comment]
Done @vfdev-5

**Pull request merged**